components/ai_chat/resources/untrusted_conversation_frame/components/context_actions_assistant/index.tsx

```diff
@@ -0,0 +1,95 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import Button from '@brave/leo/react/button'
+import Icon from '@brave/leo/react/icon'
+import { getLocale } from '$web-common/locale'
+import classnames from '$web-common/classnames'
+import { useUntrustedConversationContext } from '../../untrusted_conversation_context'
+import CopyButton from '../copy_button'
+import styles from './style.module.scss'
+
+enum RatingStatus {
+  Liked,
+  Disliked,
+  None
+}
+
+interface ContextActionsAssistantProps {
+  turnUuid?: string
+  onEditAnswerClicked: () => void
+  onCopyTextClicked: () => void
+}
+
+export default function ContextActionsAssistant(
+  props: ContextActionsAssistantProps
+) {
+  const conversationContext = useUntrustedConversationContext()
+  const [currentRatingStatus, setCurrentRatingStatus] =
+    React.useState<RatingStatus>(RatingStatus.None)
+
+  const hasSentRating = currentRatingStatus !== RatingStatus.None
+
+  function handleLikeAnswer() {
+    if (!props.turnUuid) return
+    if (hasSentRating) return
+    setCurrentRatingStatus(RatingStatus.Liked)
+    conversationContext.parentUiFrame?.rateMessage(props.turnUuid, true)
+  }
+
+  function handleDislikeAnswer() {
+    if (!props.turnUuid) return
+    if (hasSentRating) return
+    setCurrentRatingStatus(RatingStatus.Disliked)
+    conversationContext.parentUiFrame?.rateMessage(props.turnUuid, false)
+  }
```

Comment:
optional, these could be merged taking a status as a param

Reply:
Good idea! Merged as 1 method now!